dev/tasks/r/github.packages.yml

```diff
@@ -212,47 +212,62 @@ jobs:
 
   test-linux-binary:
     needs: [source, linux-cpp]
-    name: Test binary {{ '${{ matrix.image }}' }}
+    name: Test binary {{ '${{ matrix.config.image }}' }}
     runs-on: ubuntu-latest
-    container: {{ '${{ matrix.image }}' }}
+    container: {{ '${{ matrix.config.image }}' }}
     strategy:
       fail-fast: false
       matrix:
-        image:
-          - "rhub/ubuntu-gcc-release" # ubuntu-20.04 (focal)
-          - "rstudio/r-base:4.1-bionic"
-          - "rstudio/r-base:4.2-centos7"
-          - "rocker/r-ver:3.6.3" # for debian:buster (10)
-          - "rocker/r-ver" # ubuntu-20.04
-          - "rhub/fedora-clang-devel" # tests distro-map.csv, mapped to ubuntu-18.04
-          - "rocker/r-ubuntu:22.04" # tests openssl3 compatibility
+        config:
+          # If env_var is unset, we're testing that we're automatically choosing
+          # a binary on this OS. If env_var is TRUE, we're on an OS that
+          # is not in the allowlist, so we have to opt-in to use the binary
+          # Other env vars used in r_docker_configure.sh can be added here
+          # (like devtoolset) and wired up in the later steps.
+          - {image: "rhub/debian-clang-devel", env_var: "TRUE"}
+          # fedora-clang-devel cannot use binaries bc of libc++ (uncomment to see the error)
+          - {image: "rhub/fedora-clang-devel", env_var: "TRUE"}
+          - {image: "rhub/ubuntu-gcc-release"} # ubuntu-20.04 (focal)
+          - {image: "rocker/r-ubuntu:22.04"} # openssl3
+          - {image: "rocker/r-ver"} # ubuntu-20.04
+          - {image: "rocker/r-ver:3.6.3", env_var: "TRUE"} # debian:buster (10)
+          - {image: "rstudio/r-base:4.1-bionic"}
+          - {image: "rstudio/r-base:4.2-centos7"}
+          # Even with devtoolset-8, binaries fail because glibc is older than
+          # what we build with on ubuntu-18.04 (uncomment to see the error)
+          # TODO(ARROW-16976): build binaries with older glibc
+          # - {image: "rstudio/r-base:4.2-centos7", devtoolset: "8"}

```

Comment:
🙃 Oooooph this is frustrating (and sadly(?) one of the places that having the binary auto-used will help the most...)

Reply:
I think it's solvable, I'm just deferring the solution.

Reply:
Yup yup, totally